BraveRewardsTests/LedgerDatabaseTests.mm

```diff
@@ -144,6 +147,638 @@ - (void)testNonZeroExcludedCount
+  
+  XCTAssertEqual(resultWithOffset.count, 1);
+}
+
+- (void)testPublishersWithFiltersSorting
+{
+  const auto filter = [[BATActivityInfoFilter alloc] init];
+  
+  [self createBATPublisherInfo:@"1" reconcileStamp:30 percent:30 createActivityInfo:true];
+  [self createBATPublisherInfo:@"2" reconcileStamp:40 percent:40 createActivityInfo:true];
+  [self createBATPublisherInfo:@"3" reconcileStamp:10 percent:10 createActivityInfo:true];
+  
+  // Ascending order
+  const auto percentAscending = [[BATActivityInfoFilterOrderPair alloc] init];
+  percentAscending.propertyName = @"percent";
+  [percentAscending setAscending:YES];
+  
+  [filter setOrderBy:@[percentAscending]];
+  
+  const auto ascendingResult = [BATLedgerDatabase publishersWithActivityFromOffset:0
+                                                                             limit:0
+                                                                            filter:filter];
+
+  XCTAssertEqual(ascendingResult[0].id, @"3");
+  XCTAssertEqual(ascendingResult[1].id, @"1");
+  XCTAssertEqual(ascendingResult[2].id, @"2");
+  
+  // Descending order
+  const auto percentDescending = [[BATActivityInfoFilterOrderPair alloc] init];
+  percentDescending.propertyName = @"percent";
+  [percentDescending setAscending:NO];
+  
+  [filter setOrderBy:@[percentDescending]];
+  
+  const auto descendingResult = [BATLedgerDatabase publishersWithActivityFromOffset:0
+                                                                              limit:0
+                                                                             filter:filter];
+  
+  XCTAssertEqual(descendingResult[0].id, @"2");
+  XCTAssertEqual(descendingResult[1].id, @"1");
+  XCTAssertEqual(descendingResult[2].id, @"3");
+  
+  // Two sort descriptors
+  
+  // Add two more publishers with same `percent` but different `reconcileStamp`
+  [self createBATPublisherInfo:@"4" reconcileStamp:40 percent:10 createActivityInfo:true];
+  [self createBATPublisherInfo:@"5" reconcileStamp:30 percent:10 createActivityInfo:true];
+  
+  [filter setOrderBy:@[percentAscending]];
+  const auto oneSortResult = [BATLedgerDatabase publishersWithActivityFromOffset:0
+                                                                           limit:0
+                                                                          filter:filter];
+  
+  const auto oneSortResultIds = @[oneSortResult[0].id, oneSortResult[1].id, oneSortResult[2].id];
+  const auto expectedIds = @[@"3", @"5", @"4"];
+  
+  // Make sure records by `reconcileStamp` are in wrong order when using one order pair
+  XCTAssertFalse([oneSortResultIds isEqualToArray:expectedIds]);
+  
+  const auto stampAscending = [[BATActivityInfoFilterOrderPair alloc] init];
+  stampAscending.propertyName = @"reconcileStamp";
+  [stampAscending setAscending:YES];
+  
+  [filter setOrderBy:@[percentAscending, stampAscending]];
```

Comment:
[Minor/Lang]: You can call setters using dot notation, same as Swift like `stampAscending.ascending = YES;` and `filter.orderBy = @[percentAscending, stampAscending];`

Is there a reason you called the `setX` method in these few places here and above?

Reply:
i remember that Xcode complained something about it and suggested to use `setX` instead, 
it's just me fighting with objc syntax, changing it to dot notation